src/test/java/picard/analysis/CollectRnaSeqMetricsTest.java

```diff
@@ -382,6 +382,60 @@ public void testTranscriptionStrandMetrics() throws Exception {
         Assert.assertEquals(metrics.PCT_R2_TRANSCRIPT_STRAND_READS, 0.333333);
     }
 
+    @Test
+    public void testPctRibosomalBases() throws Exception {
+        final String sequence = "chr1";
+        final String ignoredSequence = "chrM";
+
+        // Create some alignments that hit the ribosomal sequence, various parts of the gene, and intergenic.
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+        // Set seed so that strandedness is consistent among runs.
+        builder.setRandomSeed(0);
+        final int sequenceIndex = builder.getHeader().getSequenceIndex(sequence);
+
+        builder.addPair("rrnaPair", sequenceIndex, 400, 500, false, false, "35I1M", "35I1M", true, true, -1);
+        builder.addFrag("frag1", sequenceIndex, 150, true,false,"34I2M","*",-1);
+        builder.addFrag("frag2", sequenceIndex, 190, true,false,"35I1M","*",-1);
+        builder.addFrag("ignoredFrag", builder.getHeader().getSequenceIndex(ignoredSequence), 1, false);
+        final File samFile = File.createTempFile("tmp.collectRnaSeqMetrics.", ".sam");
+        samFile.deleteOnExit();
+
+        final SAMFileWriter samWriter = new SAMFileWriterFactory().makeSAMWriter(builder.getHeader(), false, samFile);
+        for (final SAMRecord rec : builder.getRecords()) {
+            samWriter.addAlignment(rec);
+        }
+        samWriter.close();
```

Comment:
Since `SAMFileWriter` is `AutoCloseable` you can avoid explicit `close()` by using try-with-resources, e.g.
```
try (final SAMWriter samWriter = new SAMFileWriterFactory()...) {
    for (...) {
        ...
    }
}
```

Reply:
fixed, thx